src/wallet/scriptpubkeyman.cpp

```diff
@@ -469,6 +469,12 @@ bool LegacyScriptPubKeyMan::CanGetAddresses(bool internal) const
 bool LegacyScriptPubKeyMan::Upgrade(int prev_version, int new_version, bilingual_str& error)
 {
     LOCK(cs_KeyStore);
+
+    if (m_storage.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
```

Comment:
nit: If the user is expecting the wallet upgrade, shouldn't this condition return false and inform the reason in `error`?

```suggestion
    if (m_storage.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
            error = _("Private keys disabled");
            return false;
    }
```

Reply:
No, the wallet is still being upgraded - the version number will change, and in the future (perhaps in `CWallet`), there may actually be an upgrade. It is fine for an upgrade to do nothing - that is what we do for descriptor wallets.